vassal-app/src/main/java/VASSAL/configure/UniquelyNamedChildren.java

```diff
@@ -43,21 +41,22 @@ public UniquelyNamedChildren(AbstractConfigurable target,
   @Override
   public void validate(Buildable b, ValidationReport report) {
     if (b == target) {
-      final Map<String, String> children = new HashMap<>();
-      final List<String> duplicates = new ArrayList<>();
+      final Set<String> children = new HashSet<>();
+      final Set<String> duplicates = new HashSet<>();
 
       for (final Object child : target.getAllDescendantComponentsOf(childClass)) {
         if (child instanceof Configurable) {
+
           final String name = ((Configurable) child).getConfigureName();
-          final String exists = children.get(name);
-          if (exists == null) {
-            children.put(name, "Y");
-          }
-          else {
+
+          if (children.contains(name)) {
             if (!duplicates.contains(name)) {
               duplicates.add(name);
```

Comment:
Since `duplicates` is a `Set`, you can just `add`. There's no need to check if the item is there already.

Reply:
Thanks, I have not used Sets much before.

Reply:
Ah, You're talking about duplicates, not children. Fixing that up now.